frontend/packages/dev-console/src/components/pipelines/pipeline-resource/PipelineResourceParam.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { ActionGroup, ButtonVariant, Button } from '@patternfly/react-core';
+import { CheckIcon, CloseIcon } from '@patternfly/react-icons';
+import { useFormikContext, FormikValues } from 'formik';
+import { ButtonBar } from '@console/internal/components/utils';
+import PipelineGitOptions from './PipelineGitOptions';
+import PipelineImageOptions from './PipelineImageOptions';
+import PipelineClusterOptions from './PipelineClusterOptions';
+import PipelineStorageOptions from './PipelineStorageOptions';
+import './PipelineResourceParam.scss';
+
+export interface PipelineResourceParamProps {
+  type: string;
+}
+
+const PipelineResourceParam: React.FC<PipelineResourceParamProps> = ({ type }) => {
+  const { errors, handleReset, status, isSubmitting, dirty, submitForm } = useFormikContext<
+    FormikValues
+  >();
+  const handleCreateFormSubmit = (e: React.SyntheticEvent) => {
+    e.stopPropagation();
+    submitForm();
+  };
+  const resourceComponent = (): React.ReactElement => {
+    switch (type) {
+      case 'git':
+        return <PipelineGitOptions />;
+      case 'image':
+        return <PipelineImageOptions />;
+      case 'cluster':
+        return <PipelineClusterOptions />;
+      case 'storage':
+        return <PipelineStorageOptions />;
+      default:
+        return null;
+    }
+  };
+
+  return (
+    <div className="odc-pipeline-resource-param__content">
+      {resourceComponent()}
+      <ButtonBar errorMessage={status && status.submitError} inProgress={isSubmitting}>
+        <ActionGroup className="pf-c-form pf-c-form__actions--right pf-c-form__group--no-top-margin">
+          <Button
+            type="button"
+            variant={ButtonVariant.link}
+            onClick={handleCreateFormSubmit}
+            isDisabled={!dirty || !_.isEmpty(errors)}
+            className="odc-pipeline-resource-param__action-btn"
+            aria-label="create"
```

Comment:
This is `start` right? I am not sure about the verbiage here, but I don't think we are "creating the pipeline run" we are starting it. @christianvogt ?

Reply:
@andrewballantyne this is a sub form within the larger form and it's creating a `PipelineResource` so this is correct.

Reply:
This is for the Check icon and to create a new pipeline resource on the fly.

Reply:
Oh, right - this is the inline form not the Pipeline modal actions. nvm me